boot.go

```diff
@@ -82,8 +82,12 @@ func main() {
 				log.Printf("Starting health check server on port %d", cnf.HealthSrvPort)
 				healthSrvCh := make(chan error)
 				go func() {
-					if err := healthsrv.Start(cnf.HealthSrvPort, kubeClient.Namespaces(), storageDriver, circ); err != nil {
-						healthSrvCh <- err
+					for{
+						if err := healthsrv.Start(cnf.HealthSrvPort, kubeClient.Namespaces(), storageDriver, circ); err != nil {
+							healthSrvCh <- err
+							break
+						}
+						log.Print("namespace watcher channel closed.Starting the watch again")
```

Comment:
>do you mean exiting of the builder process

Yep

>it doesn't effect any builder flow?

Well, the builder won't function properly if the watcher dies (deleted apps won't be cleaned up), so I'd say it does have an effect on builder flow